app/Rules/IntegerIDRule.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Rules;
+
+use Illuminate\Contracts\Validation\Rule;
+
+class IntegerIDRule implements Rule
+{
+	protected bool $isNullable;
+	protected bool $isRelaxed;
+
+	public function __construct(bool $isNullable, bool $isRelaxed = false)
+	{
+		$this->isNullable = $isNullable;
+		$this->isRelaxed = $isRelaxed;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function passes($attribute, $value): bool
+	{
+		return
+			(
+				$value === null &&
+				$this->isNullable
+			) || (
+				$this->isRelaxed &&
+				filter_var($value, FILTER_VALIDATE_INT) !== false &&
+				intval($value) > 0
+			) || (
+				is_int($value) &&
```

Comment:
Legacy ID translation fails because of this condition (`is_int`). I can make it work by changing it to `is_numeric` but then I'm not sure what would be the intended difference between this case and the one right above?

Reply:
Fixed this in https://github.com/LycheeOrg/Lychee-front/pull/282/commits/cd6df802744bee4342a06c5c997935f2c6a051bc and in https://github.com/LycheeOrg/Lychee/pull/1217/commits/6c99d6a3ee6e8f5f5b3693e177b5c77ab21a07fb.

The difference between `is_int` and `is_numeric` is that the latter also excepts numbers encoded as strings. But I really wanted `is_int` now after we have proper JSON everywhere. But I missed to update the rest.